### ISSUE TITLE
can't assign function to global variable ( code execution happens inside script.runInContext)

### ISSUE THREAD

**Issue by joxerTMD** [nodejs/help#2094]
Hi i'm facing a very strange behavior and would be glad for some help or explanation

* **Node.js Version**: v12.1.0 
* **OS**:
      System Version: macOS 10.13.6 (17G2208)
      Kernel Version: Darwin 17.7.0
* **Scope (install, code, runtime, meta, other?)**: runtime
* **Module (and version) (if relevant)**:

This code fails during execution (i'm trying to assign a function inside isolated context)
```js
const vm = require('vm')
const codeString = `
'use strict';
exports = function someFunc() {}
`
const s = new vm.Script(codeString);
s.runInContext(vm.createContext({
    exports: {}
}));
```
The error is:
```
ReferenceError: exports is not defined
    at evalmachine.<anonymous>:3:9
    at Script.runInContext (vm.js:134:20)
    at Object.<anonymous>

```
looks like its somehow looses context passed with 
```
vm.createContext({
    exports: {}
})
```

however this code completes with no errors ( assign someString instead of function)...
```
const vm = require('vm')
const codeString = `
'use strict';
exports = 'someString'
`
const s = new vm.Script(codeString);
s.runInContext(vm.createContext({
    exports: {}
}));
```

**Comment by Hakerh400** [comment]
That is a known limitation. See relevant discussions: [#5344](https://github.com/nodejs/node/issues/5344), [#12300](https://github.com/nodejs/node/issues/12300), [#5679](https://github.com/nodejs/node/issues/5679). It was partially fixed in [#16487](https://github.com/nodejs/node/pull/16487), but it still does not work for functions (in [node_contextify.cc](https://github.com/nodejs/node/blob/08977822ee40b69e424b09bdb5f30fb3edbe23a8/src/node_contextify.cc#L410-L417) on line 411 - `!is_function` check). This issue is much more complex than it seems and fixing it would probably require refactoring the whole underlying logic of the `vm` module. There was a tracking issue [#6283](https://github.com/nodejs/node/issues/6283), but it has been closed with some unresolved cases.

**Comment by gireeshpunathil** [comment]
closing as answered, pls reopen if there is anything outstanding.
